Framework/API/src/Algorithm.cpp

```diff
@@ -32,6 +32,9 @@
 
 #include <map>
 
+// Index property handling template definitions
+#include "MantidAPI/Algorithm.tcc"
```

Comment:
That being said, I would be grateful if anyone could give me some advice on how to get around this problem.